datarepo-clienttests/src/main/java/scripts/testscripts/baseclasses/SimpleDataset.java

```diff
@@ -53,6 +63,11 @@ public void setup(List<TestUserSpecification> testUsers) throws Exception {
   }
 
   public void cleanup(List<TestUserSpecification> testUsers) throws Exception {
+    cleanup(testUsers, true);
+  }
+
+  public void cleanup(List<TestUserSpecification> testUsers, boolean deleteProfile)
```

Comment:
If it's this class's responsibility to clean up the created profile (since it was this class that created it), then it should be this class's responsibility to track this state, not the caller. Such as by initializing a boolean field in `setup()` based on `billingProfile == null`.

Reply:
Yeah, it's a fair point.  I'll pull the things into one class.  the entry points to the super class make it a little tough but I'll see if I can clean it up a bit